src/processSources.ts

```diff
@@ -0,0 +1,107 @@
+import type { Source } from "@graphql-tools/utils";
+import crypto from "crypto";
+import { OperationDefinitionNode, FragmentDefinitionNode } from "graphql";
+
+export type OperationOrFragment = {
+  initialName: string;
+  definition: OperationDefinitionNode | FragmentDefinitionNode;
+};
+
+export type SourceWithOperations = {
+  source: Source;
+  operations: Array<OperationOrFragment>;
+};
+
+const expandSources = (sources: Array<Source>): Array<Source> => {
+  const expandedSources: Array<Source> = [];
+  for (const source of sources) {
+    const document = source.document!;
+    const rawSDL = source.rawSDL!;
+    const operationSDL = rawSDL.split(`\n#-#\n`);
+    operationSDL.forEach((rawSDL, index) => {
+      expandedSources.push({
+        document: {
+          kind: "Document",
+          definitions: [document.definitions[index]],
+        },
+        rawSDL,
+        location: source.location,
+      });
+    });
+  }
+  return expandedSources;
+};
```

Comment:
Here we split our operations that got concatenated into a single string via CodeFileLoader into back into single operations. SO we can properly create the operations raw string -> type document node mapping